### PR TITLE
update `/eth/v1/beacon/states/{state_id}/finality_checkpoints` to select state more efficiently

### PR DESCRIPTION
**Pull request by rolfyone**
Because states were always being translated to slot, there were efficiencies to be gained by making better use of stored states.

There are still times that the slot needs to be used, states like 'head' and 'finalized' are readily accessible and should be retrieved as efficiently as possible.

Signed-off-by: Paul Harris <paul.harris@consensys.net>

## Documentation

- [X] I thought about documentation and added the `documentation` label to this PR if updates are required.

### CODE REVIEW

File: data/beaconrestapi/src/test/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateFinalityCheckpointsTest.java [hunk 1]
```diff
@@ -28,42 +26,31 @@
...
-  @Test
-  public void shouldReturnUnavailableWhenStoreNotAvailable() throws Exception {
-    final GetStateFinalityCheckpoints handler =
-        new GetStateFinalityCheckpoints(chainDataProvider, jsonProvider);
-    when(context.pathParamMap()).thenReturn(Map.of("state_id", "head"));
-    when(chainDataProvider.isStoreAvailable()).thenReturn(false);
-
-    handler.handle(context);
-    verifyStatusCode(SC_NOT_FOUND);
-  }
```
Comment (ajsutton) [inline data/beaconrestapi/src/test/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateFinalityCheckpointsTest.java:46]: Assuming we don't need this because testing for pre-genesis is now just covered by the state selector which would return empty.

**Review by ajsutton — APPROVED** [review]
LGTM.

File: data/beaconrestapi/src/test/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateFinalityCheckpointsTest.java (hunk 1 above)
Comment (rolfyone) [reply data/beaconrestapi/src/test/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateFinalityCheckpointsTest.java:46]: That's my belief 👍
